wasm/src/context.c

```diff
@@ -18,6 +21,33 @@ void opa_eval_ctx_set_input(opa_eval_ctx_t *ctx, opa_value *v)
     ctx->input = v;
 }
 
+WASM_EXPORT(opa_eval_ctx_one_off)
+char *opa_eval_ctx_one_off(int entrypoint, opa_value *data, uint32_t heap, size_t input_len)
+{
+    opa_heap_ptr_set(heap);
+    opa_eval_ctx_t ctx = { .entrypoint = entrypoint, .data = data };
+
+    char *input = opa_malloc(input_len * sizeof(input));
+    if (opa_host_input(input) != 0) {
```

Comment:
This inversion is based on inspiration from waPC: Instead of having functions call in to
1. malloc and write data into the returned location
2. value_parse
3. set the input addr to what's returned by (2.),

we're instead passing the length of the input we'd like to use, and the wasm module allocates space, and informs the host about the location via `opa_host_input`. The host is expected to, in that function call, write its input into that location.